pkg/ansible/controller/controller.go

```diff
@@ -84,4 +102,30 @@ func Add(mgr manager.Manager, options Options) {
 		log.Error(err, "")
 		os.Exit(1)
 	}
+	return &c
+}
+
+func NewControllerMap() *ControllerMap {
+	return &ControllerMap{
+		internal: make(map[schema.GroupVersionKind]controller.Controller),
+	}
+}
+
+func (cm *ControllerMap) Get(key schema.GroupVersionKind) (controller controller.Controller, ok bool) {
+	cm.RLock()
+	result, ok := cm.internal[key]
+	cm.RUnlock()
```

Comment:
NIT: switch this to `defer cm.RUnlock()` above line 116

Same as the other lock and unlocks below